### ISSUE TITLE
Encoding of association lists

### ISSUE THREAD

**Issue by mmontone** [equill/neo4cl#26]
Hello again.

I'm passing some nested data to my queries, in the form of association lists.

In current implementation, an alist is considered one when cdr of all elements is an atom. That prevents me from passing nested alists, or other complex values as lists, etc.

I've relaxed that constraint removing the `(atom (cdr pair))` in `alist-p`:

```lisp
(defun alist-p (lst)
  "Check whether the argument is actually an alist."
  (and (listp lst)
       (or (null lst)
           (every #'(lambda (pair)
                      (and (consp pair)
                           (atom (car pair))
                           ;;(atom (cdr pair))
			   ))
                  lst))))
```

I'm being able to pass nested association lists now. Encoding works fine.

**Comment by equill** [comment]
Hi! Sorry for the delay in responding.
I remember noting that this was an issue, but apparently I didn't create a separate ticket to address it. So thanks for this reminder :)

This stands a chance of fixing #20 as well. I'll go back to the spec, and try to come up with a version that works for all valid cases, but excludes lists that are not dotted pairs.

**Comment by mmontone** [comment]
Hi. Ok. Do you see a problem with what I've done?

**Comment by mmontone** [comment]
Ah. Perhaps also fix #20 in the process ..

**Comment by equill** [comment]
I still haven't come up with a better test, and even the spec isn't that specific, so I'm using your solution and just dropping the requirement that the `cdr` be an atom.
